substrate/client/rpc/src/state/state_full.rs

```diff
@@ -166,6 +166,36 @@ where
 		}
 		Ok(())
 	}
+
+	/// Helper function to check a key against lists of prefixes
+	fn is_target_key(
+		&self,
+		key_to_check: StorageKey,
+		include_prefixes : Option<Vec<StorageKey>>,
+		exclude_prefixes : Option<Vec<StorageKey>>
+	) -> bool {
```

Comment:
One optimisation is needed here.
Need to pass the args to the function as read-only references - to avoid `.clone()` of params for each storage key.